src/binary/merkle_tree.rs

```diff
@@ -210,6 +247,47 @@ mod test {
         assert_eq!(s_node_3.hash(), &node_3);
     }
 
+    #[test]
+    fn load_returns_a_valid_tree() {
+        const LEAVES_COUNT: u64 = 7;
+
+        let mut storage_map = StorageMap::<u64, Node>::new();
+
+        let root_1 = {
+            let mut tree = MT::new(&mut storage_map);
+            let data = &TEST_DATA[0..LEAVES_COUNT as usize];
+            for datum in data.iter() {
+                let _ = tree.push(datum);
+            }
+            tree.root().unwrap()
+        };
+
+        let root_2 = {
+            let mut tree = MT::load(&mut storage_map, LEAVES_COUNT).unwrap();
+            tree.root().unwrap()
+        };
+
+        assert_eq!(root_1, root_2);
+    }
+
+    #[test]
+    fn load_returns_a_load_error_if_the_storage_is_not_valid_for_the_leaves_count() {
+        let mut storage_map = StorageMap::<u64, Node>::new();
+
+        {
+            let mut tree = MT::new(&mut storage_map);
+            let data = &TEST_DATA[0..5 as usize];
```

Comment:
nit: The `as` cast could be replaced with `0..5_usize`, which is more idiomatic and a bit cleaner

Reply:
Thanks for the catch - I must have been trying to cast a variable at one point, but I can just do `[0..5]` without any cast at all!